web/src/hooks/transaction/useSubmitTransaction.ts

```diff
@@ -1,18 +1,19 @@
-import { useEffect, useState } from 'react';
+import { useEffect, useMemo, useState, useRef } from 'react';
 import processViemContractError from '@/utils/processViemContractError';
-import { Abi, DecodeEventLogReturnType, TransactionReceipt, decodeEventLog } from 'viem';
 import {
-  UseSimulateContractParameters,
-  useSimulateContract,
-  useWaitForTransactionReceipt,
-} from 'wagmi';
-import { useWriteContracts, useCallsStatus } from 'wagmi/experimental';
-
-const zerodevApiKey = process.env.NEXT_PUBLIC_ZERODEV_API_KEY;
-const ZERODEV_PAYMASTER_URL = `https://rpc.zerodev.app/api/v3/paymaster/${zerodevApiKey}`;
+  Abi,
+  DecodeEventLogReturnType,
+  TransactionReceipt,
+  decodeEventLog,
+  encodeFunctionData,
+  http,
+} from 'viem';
+import { usePasskeyAccount } from '@/providers/PasskeyProvider';
+import { useWaitForTransactionReceipt } from 'wagmi';
+import { getChainsForEnvironment } from '@/store/supportedChains';
 
 const getEvents = (
-  contractCallConfig: UseSimulateContractParameters,
+  contractCallConfig: any,
```

Comment:
Will it cause any issue to remain using UseSimulateContractParameters for type safety?